### ISSUE TITLE
Question about arrays of `Orbit`s

### ISSUE THREAD

**Issue by TomWagg** [GalacticDynamics/galax#745]
Hey @nstarman! I've nearly got `galax` working entirely with `cogsworth` now 🎉 The one holdup I have is that, after computing the orbits for each binary, I tried to save them all to a single `numpy` array but then the code basically just hangs. I've tested this for even just an array of two orbits, each with just 3 timesteps, and it didn't finish after 3 minutes. I tried it with `jnp.array` too but that just gave me an error.

The reason why I want them like this is that then I can mask them easily from `cogsworth` `Population`s. For example, if I create a mask that grabs every X-ray binary based on the stellar evolution, I want to immediately be able to slice my list of orbits.

Do you have any recommendations of how I should go about doing this? (I'm also curious as to why this takes so long to run)

p.s. here's a quick example in case you want to run it
```
import jax.numpy as jnp

import unxt as u
import galax.coordinates as gc
import galax.dynamics as gd
import galax.potential as gp
import numpy as np

# start at the sun
w = gc.PhaseSpaceCoordinate(
    q=u.Quantity([8, 0, 0], "kpc"),
    p=u.Quantity([0, 220, 0], "km/s"),
    t=u.Quantity(0, "Myr"),
)
usys = u.unitsystem("kpc", "Msun", "Myr", "rad")
pot = gp.NFWPotential(
    m=u.Quantity(1.77e12, "Msun"),
    r_s=u.Quantity(20, "kpc"),
    units=usys,
)
orbit = gd.evaluate_orbit(pot, w, u.Quantity(jnp.linspace(0, 20, 3), "Myr"))

# combine orbit into list and then array
orbit_list = [orbit, orbit]
orbit_array = np.array(orbit_list)                       # <--- hangs here
```

**Comment by nstarman** [comment]
I've never tried a list of orbits, only an orbit of lists (if that makes sense).

```python
# start at the sun
w = gc.PhaseSpaceCoordinate(
    q=u.Quantity([[8, 0, 0], [9, 0, 0]], "kpc"),   # note multiple positions
    p=u.Quantity([0, 220, 0], "km/s"),
    t=u.Quantity(0, "Myr"),
)
usys = u.unitsystem("kpc", "Msun", "Myr", "rad")
pot = gp.NFWPotential(
    m=u.Quantity(1.77e12, "Msun"),
    r_s=u.Quantity(20, "kpc"),
    units=usys,
)
orbit = gd.evaluate_orbit(pot, w, t=u.Quantity(jnp.linspace(0, 20, 3), "Myr"))

orbit = jax.block_until_ready(orbit)
print(orbit.shape)
```

If you want to construct an array from a single orbit, how about then broadcasting it within the `orbit` structure?

```python
from dataclasses import replace

w = gc.PhaseSpaceCoordinate(
    q=u.Quantity([8, 0, 0], "kpc"),
    p=u.Quantity([0, 220, 0], "km/s"),
    t=u.Quantity(0, "Myr"),
)
orbit = gd.evaluate_orbit(pot, w, t=u.Quantity(jnp.linspace(0, 20, 3), "Myr"))

mask_shape = (5, 7)
shaper = jnp.ones((*mask_shape, 3))
orbit = replace(orbit, q=jax.tree.map(lambda x: x * shaper, orbit.q),
                p=jax.tree.map(lambda x: x * shaper, orbit.p))

orbit = jax.block_until_ready(orbit)
```

**Comment by nstarman** [comment]
Found a bug with ND orbit slicing. Will fix, then this will work in general. 1D lists should already work.

**Comment by TomWagg** [comment]
Thanks for the quick reply! One complication that I didn't mention: these lists are ragged. Each binary is formed at a different time in the history of the galaxy, so the length of its orbit is different. This made it tricky back in the day to combine orbit information into a single object, that's why we went with the list of orbits.

Does your method still work if the orbits have different lengths (I think no right?)?

**Comment by nstarman** [comment]
Yes, that does complicate things! Padding to make the lengths equal?
If the list is 1D just use a tuple?
